### PR TITLE
fix: allow S1 controllers to book

### LINKED ISSUE #483: Booking as finished S1
When a student has finished their S1 training and have a permanent endorsement, does not have access to book positions anymore

### PR DESCRIPTION
**Pull request by thor**
Fixes https://github.com/Vatsim-Scandinavia/controlcenter/issues/483.

* Changes the booking controller to check for active infinite
  endorsements for S1 students. This isn't pretty, but it'll do for now.

* Changes the booking policy to allow for active infinite endorsements
  for S1 students. This, in turn, allows us to see the booking widget in
  Control Center booking page.

* VERIFY: Changes the ATC activity job to also allow for something(?).

* Add tests to verify success of bookings. This uses both the HTML
  client to verify a position of the rating is available and the model
  to verify we successfully add a booking.

* Split the VatsimRating enum out in order to make it PSR-4 compliant.

### CODE REVIEW

**Comment by thor** [comment]
I haven't given this the pre-review check I'd like to, but have a look and see which dumb things we can cull quick and dirty.

**Comment by thor** [comment]
> I haven't given this the pre-review check I'd like to, but have a look and see which dumb things we can cull quick and dirty.

☝️ @blt950

**Comment by blt950** [comment]
> > I haven't given this the pre-review check I'd like to, but have a look and see which dumb things we can cull quick and dirty.
> 
> ☝️ @blt950

I haven't forgot this, just haven't got the time to check yet 😅

**Comment by thor** [comment]
> > > I haven't given this the pre-review check I'd like to, but have a look and see which dumb things we can cull quick and dirty.
> > 
> > ☝️ @blt950
> 
> I haven't forgot this, just haven't got the time to check yet 😅

Ah, no worries! I wasn't sure if you got pull request review request notifications (whoohee, what a mouthful), so I'll keep that in mind for the future!

File: app/Policies/BookingPolicy.php [hunk 1]
```diff
@@ -29,7 +31,9 @@ public function view()
      */
     public function create(User $user)
     {
-        return $user->rating >= 3 || $user->getActiveTraining(1) != null || $user->isModeratorOrAbove();
+        return ($user->active && $user->rating >= VatsimRating::S1->value
```
Comment (blt950) [inline app/Policies/BookingPolicy.php:34]: This allows all S1's to see the booking "widget" on the index page, but that's not necessary if they don't have an active training or endorsement to book a position.

Tested with a user with no endorsements, S1 rating and no active trainings. Saw this widget, could write in all information, but then 403'ed on submit. Better just to hide this all together. Except for those who already are in training or have a infinite S1 endorsement

**Comment by blt950** [comment]
I've added a review of what I managed to test atm. I'll test rest later
- Check the review
- Could you merge this into v3.4.0 branch instead? To check with conflicts in the new version instead of current release?
@thor

**Review by blt950 — CHANGES_REQUESTED** [review]
I've added a review of what I managed to test atm. I'll test rest later
- [ ] Check the review
- [ ] use App\Helpers\VatsimRating change is not reflected in `D:\GitHub\controlcenter\app\Http\Controllers\EndorsementController.php` and `D:\GitHub\controlcenter\app\Models\Handover.php` throwing errors e.g. when deleting endorsement
- [ ] When attempting creating a new manual training, I get the error `Unable to acquire an instance of Handover class for user 10000012` (MissingHandoverObjectException). Try reproducing this for more detailed errors perhaps?
- [ ] Could you merge this into v3.4.0 branch instead? To check with conflicts in the new version instead of current release?
@thor

**Comment by blt950** [comment]
I've updated my review now @thor so you can take a look and tweak what's needed :) Let me know when you're ready for next review

File: app/Policies/BookingPolicy.php (hunk 1 above)
Comment (thor) [reply app/Policies/BookingPolicy.php:34]: Good catch. Going for a simple approach with two different checks.

**Comment by thor** [comment]
* [x]  Check the review 
 * [x]  use App\Helpers\VatsimRating change is not reflected in `D:\GitHub\controlcenter\app\Http\Controllers
\EndorsementController.php` 
 * [x]  Could you merge this into v3.4.0 branch instead? To check with conflicts in the new version instead of current release?

These should be good-er-to-go.

> * [ ] `D:\GitHub\controlcenter\app\Models\Handover.php` throwing errors e.g. when deleting endorsement
> * [ ]  When attempting creating a new manual training, I get the error `Unable to acquire an instance of Handover class for user 10000012` (MissingHandoverObjectException). Try reproducing this for more detailed errors perhaps?

I was unable to reproduce this with my local MySQL setup. Could you provide additional steps to reproduce this? I was able to successfully delete a solo endorsement without any site errors.

**Review by blt950 — APPROVED** [review]
All good now! Can't wait for the bugs to appear in production

**Pull request merged**